### DISCUSSION TITLE
Installation issues... miniconda possible?

### DISCUSSION THREAD

**Discussion by zacforsman** [cnr-isti-vclab/TagLab#30] [Ideas]
I've been working for several days to install TagLab on both Windows and Ubuntu.  I managed to get the cpu version working in ubuntu wsl (win 10 and win 11 with new graphics integration) and was able to install cuda versions in both windows and ubuntu; however, many of the tools don't work and or it crashes.  

I really want to use this tool for my research but I'm having a hard time installing it, probably due to version conflicts with dependencies.  Is it possible to put this project to a container (such as miniconda) so that installation would be more foolproof?

I can't wait to work with this software!

-Zac

**Comment by maxcorsini** [comment]
Hi Zac. Please, write us the model of the NVIDIA Graphics Board and the GPU RAM: Yes, you can create an environment in conda and use it to avoid potential conflicts.

**Comment by alemuntoni** [comment]
Hi, sorry for the late reply.
Actually, there is the idea in the future to make a proper package of taglab that will be easy to install, however making it properly is not as easy as it might seem, due to several dependencies we have (mainly torch which could depend on cuda, and gdal, which in some OSs depends on external libraries that need to be installed outside python). 
It's a big TODO we have in our list, since it would require lot of time and effort.
The `install.py` script should work fine most of the times, but I see that there could be some system configurations that may break the whole installation of all the dependencies.

**Comment by zacforsman** [comment]
Thanks for the assistance,
   I did manage to install and get the program running... After I found the
dependencies file I used the archived windows files and it worked well so
far.    I am currently testing it out now and very grateful to have this
software available. Please keep up the great work.
-Zac




On Tue, Mar 1, 2022 at 7:03 PM Alessandro Muntoni ***@***.***>
wrote:

> Hi, sorry for the late reply.
> Actually, there is the idea in the future to make a proper package of
> taglab that will be easy to install, however making it properly is not as
> easy as it might seem, due to several dependencies we have (mainly torch
> which could depend on cuda, and gdal, which in some OSs depends on external
> libraries that need to be installed outside python).
> It's a big TODO we have in our list, since it would require lot of time
> and effort.
> The install.py script should work fine most of the times, but I see that
> there could be some system configurations that may break the whole
> installation of all the dependencies.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cnr-isti-vclab/TagLab/discussions/30#discussioncomment-2274358>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADDIC3G7YMJ6FFAHFL6VBUTU5Y5TLANCNFSM5PIIN3NA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Zac H. Forsman,  Ph.D.
Assistant Researcher | Coral Conservation Genetics & Restoration
Hawai'i Institute of Marine Biology | University of Hawai'i, Mānoa
Coconut Island | Moku o Loʻe | 46-007 Lilipuna Road | Kāne'ohe, HI 96744
https://zacforsman.github.io/

**Comment by zytoh0** [comment]
I've installed successfully but when I try to run `python3 TagLab.py`, `ModuleNotFoundError: No module named '_gdal'` came out.
I've checked `gdal-config --version` and it shows `3.1.3`. I'm using `NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"`

**Comment by maxcorsini** [comment]
Hi Zytoh0. Let's try to follow the suggestions contained in the following thread:

https://stackoverflow.com/questions/44005694/no-module-named-gdal

I suggest to you to begin from the renaming of the library under the osgeo directory in _gdal.so 

Hope this helps !

**Comment by landlala9** [comment]
Hi there!
I'd also love to use your software, but I am not able to install it because when I try to run `python3 install.py` it shows me this message:

Torch will be installed in its CPU version.
GDAL version installed: 3.5.1
Trying to install cmake...
Traceback (most recent call last):
  File "install.py", line 180, in <module>
    raise Exception('Error while configuring Coraline library.\nInstallation aborted.')
Exception: Error while configuring Coraline library.
Installation aborted.

Any ideas what I could do to solve this problem?

**Reply by alemuntoni** [reply]
Are you using MacOS or Linux?
In any case, as described in the [Readme](https://github.com/cnr-isti-vclab/TagLab#step-0-dependencies), be sure to have apt (linux) or homebrew (macos), in order to allow the installation of cmake and gdal.

**Reply by landlala9** [reply]
I am using MacOS, but I do have installed homebrew
